### DISCUSSION TITLE
Enable https

### DISCUSSION THREAD

**Discussion by PGLange** [Budibase/budibase#1772] [Help] [answered]
Hi there,
budibase is great! I like it very much, and I'd like to use it outside my LAN. Therefor I need to setup HTTPS, I have a valid lets-encrypt certificate. Using Ubuntu and Docker-Setup (selfhosting but not on digital ocean) -
Can you give me a hint, how to let Budibase containers know where the certiftcate lives?

**Comment by aptkingston** [comment]
Hey @PGLange. Since Budibase is primarily for developing internal tools and is expected to be typically run inside a secure LAN, HTTPS support isn't something we currently ship as standard. If you'd like external access then you would be better looking at running your own reverse proxy, setting up HTTPS and forwarding traffic to your Budibase install. We have some docs about that [here](https://docs.budibase.com/self-hosting/advanced-options/reverse-proxy).

If you've got a docker environment up and running already then I can personally recommend linuxserver's [SWAG](https://hub.docker.com/r/linuxserver/swag) container, which bundles Nginx for a reverse proxy and Lets Encrypt for certificate automation.

**Reply by oswaldB** [reply]
Hello personnaly, I use caddy as a reverse proxy and it is working perfectly.

Once my installation done, I install the caddy and then run do a 'caddy stop' ( I don't know why but it prevent some errors) and then I run caddy reverse-proxy --from mydomainename.com --to localhost:1000 &

I hope it can help you

**Reply by PGLange** [reply]
Hi Oswald,
that is too easy :) :) :) THANK YOU!!!! 
@aptkingston  Thank you for your very fast answer, I love your Idea, but I ran into deep troubles trying this out. I will definetly spend some time on this as the whole setup fits in one docker-compose file :)

**Reply by HideoKuze** [reply]
@oswaldB Hey did you install Caddy for docker or just linux? Im still having trouble doing this, it tells me port 80 is already in use (by Budibase) when I just install it for CentOS or the Docker image. I'm not sure how everyone else is getting this to work so easily

**Comment by oswaldB** [comment, marked as answer]
Did you succeed?

I am happy to see it helps you.

**Reply by PGLange** [reply]
Ay! And it works fine... just download, stopped the "new" caddy service and "restartet" it with:
`caddy reverse-proxy --from mydomainename.com --to localhost:<<my_port>> &`

**Reply by enpro-github** [reply]
Hey! Could you help me with this by any chance? I have Budibase hosted on an EC2 instance in AWS. But when I run `caddy reverse-proxy --from <ipaddress>.us-eat-2.compute.amazonaws.com --to localhost:1000` it eventually times out. I see an error that says: `could not get certificate from issuer`

**Reply by aptkingston** [reply]
@enpro-github not sure if that's just a typo from writing it here, but that should be `us-east` not `us-eat` - in case the issue was as simple as that!

**Reply by enpro-github** [reply]
@aptkingston Hey thanks for the reply! That was a type sorry, I'm still having the problem. Is there not anything else I need to do? Like download a SSL certificate? I'm confused as to why this is so "easy"

**Reply by enpro-github** [reply]
@aptkingston I actually opened a discussion here: https://github.com/Budibase/budibase/discussions/4536

**Comment by joebudi** [comment]
@oswaldB Caddy is a great suggestion and we will add it to our docs. If you like, I can send you a nice Budibase t-shirt and some merch as a way of saying thank you?

**Comment by oswaldB** [comment]
Hello,

With pleasure, but I am more than happy to help you to build an amazing
no-code open source program that fits with the European requirements on
data!



On Mon, 21 Jun 2021 at 13:34, Joe ***@***.***> wrote:

> @oswaldB <https://github.com/oswaldB> Caddy is a great suggestion and we
> will add it to our docs. If you like, I can send you a nice Budibase
> t-shirt and some merch as a way of saying thank you?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Budibase/budibase/discussions/1772#discussioncomment-897556>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAXCFJVOSGJZUVYK5FS4A4DTT4WWJANCNFSM47A62Q6Q>
> .
>


-- 
Oswald BERNARD
00 44 7861004200

**Comment by shogunpurple** [comment]
Caddy HTTPS setup now added to our docs.

https://docs.budibase.com/self-hosting/advanced-options/reverse-proxy

Thanks very much for this, it should help budibase users get set up with HTTPS much more quickly!

**Comment by Tobiaswec** [comment]
Hi im running the budibase-aas image on an Azure Container Instance, i want to ensure HTTPS by using Caddy as separate ACI but i have trouble forwarding to Budibase since its running on port 80 and i was not able to change that, how can i setup Caddy/Budibase correctly

Caddyfile:
MYAPP.westeurope.azurecontainer.io {
    reverse_proxy   http://localhost:PORT
}

**Reply by Tobiaswec** [reply]
Another Option i considered was using an Azure App Service this provides SSL out of the box but i cant not persist data onto another Storage like Azure File Share
